src/api/Credex/services/AcceptCredex.ts

```diff
@@ -27,23 +27,53 @@ export async function AcceptCredexService(
 ): Promise<AcceptCredexResult | null> {
   logDebug(`Entering AcceptCredexService`, { credexID, signerID, requestId });
 
-  if (!credexID || !signerID) {
-    logError("AcceptCredexService: credexID and signerID are required", new Error("Missing parameters"), { credexID, signerID, requestId });
+  if (!credexID || !signerID || !requestId) {
+    logError("AcceptCredexService: Missing required parameters", new Error("Missing parameters"), { credexID, signerID, requestId });
     return null;
   }
 
   const ledgerSpaceSession = ledgerSpaceDriver.session();
 
   try {
-    logDebug(`Attempting to accept Credex in database`, { credexID, signerID, requestId });
+    logDebug(`Attempting to accept Credex in database`, { 
+      credexID, 
+      signerID, 
+      requestId,
+      sessionState: ledgerSpaceSession.lastBookmark() 
+    });
 
     const result = await ledgerSpaceSession.executeWrite(async (tx) => {
+      // First, verify the nodes exist
+      const verifyQuery = `
+        MATCH (credex:Credex {credexID: $credexID})
+        MATCH (signer:Member|Avatar {memberID: $signerID})
+        RETURN 
+          credex IS NOT NULL as credexExists,
+          signer IS NOT NULL as signerExists
+      `;
+      
+      const verifyResult = await tx.run(verifyQuery, { credexID, signerID });
+      
+      if (verifyResult.records.length === 0) {
+        logError("Verification failed - Credex or Signer not found", new Error("Entities not found"), {
+          credexID,
+          signerID,
+          requestId
+        });
+        return null;
+      }
+
       const query = `
         MATCH
           (issuer:Account)-[rel1:OFFERS]->
           (acceptedCredex:Credex {credexID: $credexID})-[rel2:OFFERS]->
           (acceptor:Account)<-[:AUTHORIZED_FOR]-
           (signer:Member|Avatar { memberID: $signerID })
+        WITH issuer, rel1, acceptedCredex, rel2, acceptor, signer
```

Comment:
this must be redundant. the patterns matched above would not return results that could possibly be null in any of these fields